src/OSPSuite.Core/Domain/Services/ParameterIdentifications/ParameterIdentificationRun.cs

```diff
@@ -229,10 +235,14 @@ private IReadOnlyList<SimulationRunResults> runSimulations()
 
       private void updateParameterValues(IReadOnlyList<OptimizedParameterValue> values)
       {
-         if (_parameterIdentification.AllIdentificationParameters.Count != values.Count)
-            throw new ArgumentException(Error.IdentificationParametersAndValuesDoNotTheSameLength(_parameterIdentification.AllIdentificationParameters.Count, values.Count));
+         if (_variableParameters.Count != values.Count)
+            throw new ArgumentException(Error.IdentificationParametersAndValuesDoNotTheSameLength(_variableParameters.Count, values.Count));
 
-         _parameterIdentification.AllIdentificationParameters.Each((x, i) => addParameterValue(x, values[i].Value));
+         //Update variable parameters according to optimized parameter values
+         _variableParameters.Each((x, i) => addParameterValue(x, values[i].Value));
+
+         //Preserve fixed parameter values
+         _fixedParameters.Each((x, i) => addParameterValue(x, x.StartValue));
```

Comment:
I thought about that but to be honest, I am not sure that this will bring any significant speed improvement as the number of parameters affected by that will probably be very small anyways